### DISCUSSION TITLE
Camara QoD API Device/UE Private IP

### DISCUSSION THREAD

**Discussion by ajmhussain** [camaraproject/Commonalities#288] [Q&A]
We are in the process of the planning to proceed with the subject API and discussions are already ongoing with internal stakeholders from network side. For Create Session API, we need some clarification regarding device/UE Private IP as it is defined as optional parameter in your specifications, but we expect it from the API consumer itself. We couldn’t find any other feasible option from our discussion with the network team, and they are also of the view that public and private IP pair has to be passed by the API consumer and same should be sent to the backend network to establish the QoD session.

Can you please advise on this and a clarification that why the parameters are set as optional in Camara APIs.

**Comment by rartych** [comment]
Device identification parameters are set optional since API consumers can use the authenticated device identifier associated with 3-legged access tokens see the explanation in https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#appendix-a-infodescription-template-for-device-identification-from-access-token
